frame/system/src/lib.rs

```diff
@@ -589,8 +600,17 @@ decl_module! {
 		/// # <weight>
 		/// - `O(I)` where `I` length of `items`
 		/// - `I` storage writes (`O(1)`).
+		/// - Base Weight: 0.568 * i
+		/// - Writes: Number of items
 		/// # </weight>
-		#[weight = (0, DispatchClass::Operational)]
+		#[weight = FunctionOf(
+			|(items,): (&Vec<KeyValue>,)| {
+				T::DbWeight::get().writes(items.len() as Weight)
+					.saturating_add((items.len() as Weight).saturating_mul(600_000))
+			},
+			DispatchClass::Operational,
+			Pays::Yes,
```

Comment:
Do we still use FunctionOf ? I thought the new standard was to use tuple when you want to give more information:

```
(
T::DbWeight::get().writes(items.len() as Weight).saturating_add((items.len() as Weight).saturating_mul(600_000)),
DispatchClass::Operational
)
```
Pays::Yes is default

Reply:
You need function of to access the parameters, right?

Reply:
It is also my understanding that the tuple can only be used when your weight does not depend on the arguments of the function.

Reply:
Field are accessible https://github.com/paritytech/substrate/pull/5828/files#diff-2e0725fa587f558456a6b465bbd8dc0bR1131
But maybe I'm abusing the current macro generation